pandas/core/dtypes/common.py

```diff
@@ -737,5 +737,7 @@ def pandas_dtype(dtype):
             pass
     elif isinstance(dtype, ExtensionDtype):
         return dtype
+    elif np.dtype(dtype).kind == 'O':
```

Comment:
let's add a few more tests cases, FYI (e.g. ``np.float64``, 'float64', np.dtype('float64')``, IOW some objects and strings

maybe something like

``
if isinstance(dtype, (np.dtype, compat.string_types)):
    return np.dtype(dtype)
raise TypeError
```